sled-agent/types/src/rack_ops.rs

```diff
@@ -44,3 +44,57 @@ pub enum RackOperationStatus {
         id: RackResetUuid,
     },
 }
+
+/// Steps we go through during initial rack setup.
+/// Keep this list in order that they happen.
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Deserialize,
+    EnumCount,
+    EnumIter,
+    Eq,
+    Hash,
+    JsonSchema,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+)]
+#[serde(tag = "status", rename_all = "snake_case")]
+pub enum RssStep {
+    Requested,
+    Starting,
+    CheckPrevious,
+    LoadExistingPlan,
+    WaitingForPeers,
+    InitTrustQuorum,
+    NetworkConfigUpdate,
+    SledInit,
+    EnsureStorage,
+    ConfigureDNS,
+    InitDNS,
+    InitNTP,
```

Comment:
Tiny style nit - I think these should be 
```suggestion
    ConfigureDns,
    InitDns,
    InitNtp,
```